Retrieve viz url

Huongg opened the pull request:
## Description
Fixes #1851 
Design is [here]( https://www.figma.com/file/3kSpvIO1veLKfy9qHxuXwF/Kedro-WIP?type=design&node-id=1411-87299&mode=design&t=bSwcXxbq95GPPfud-4)
![retrieve-url](https://github.com/kedro-org/kedro-viz/assets/32060364/086160e8-183d-4ccb-9103-0f73be427079)





## Development notes

- we store store the fields including hosting platform, bucket name and endpoint URL in [localStorage](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage) once the user publishes it. The format is as follow
```
{
   aws: {
       bucket_name: ...,
       platform: 'aws',
       endpoint: '....',
   },
   gcp: {....},
   azure: {....}
}
```

- Render the published content view will depends on whether there's only 1 hosting platform stored in localStorage or more than 1
- Current logic is summarised as the chart below

<img width="554" alt="Screenshot 2024-05-21 at 16 48 53" src="https://github.com/kedro-org/kedro-viz/assets/32060364/56b6cfae-2d10-4251-9b72-4711cd9c65b6">

## QA notes
You can test the UI using Gitpod. I've implemented the logic prior to the check that validates whether the submitted platform is valid or not. As a result, you can use any fabricated names and URLs, the system will display an error message.

<!-- How has the expected behaviour changed? What testing strategies have you used? -->

## Checklist

- [ ] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [x] Added new entries to the `RELEASE.md` file
- [x] Added tests to cover my changes

rashidakanchwala commented on src/components/shareable-url-modal/shareable-url-modal.js:
```diff
@@ -4,35 +4,37 @@ import { connect } from 'react-redux';
     success:
-      'The current version of Kedro-Viz has been published and hosted via the link below.',
+      'The deployment has been successful and Kedro-Viz is hosted via the link below..',
     incompatible: `Publishing Kedro-Viz is only supported with fsspec>=2023.9.0. You are currently on version ${info}.\n\nPlease upgrade fsspec to a supported version and ensure you're using Kedro 0.18.2 or above.`,
```
Still think these hard coded messages should be in config.js and not here.

Huongg replied:
done, moved it to config

jitu5 commented on src/components/shareable-url-modal/shareable-url-modal.scss:
```diff
@@ -91,29 +103,49 @@
     }
   }
 
+  &__content-description-title:nth-of-type(2)::before {
```
Which element we are trying to select here? Is it possible to avoid nth-of-type and use targeted class instead ?

Huongg replied:
yup good point, let me do that 👍

jitu5 commented on src/components/shareable-url-modal/shareable-url-modal.scss:
```diff
@@ -275,6 +330,75 @@
+  width: 100%;
+
+  .button__btn--secondary {
+    // double check the hex colour with Step
```
Please remove comment if not needed.

Huongg replied:
I'll leave the comment here so when Stephanie is QA'ing the code I can remember to check with her. Will remove before merging

rashidakanchwala commented on src/components/shareable-url-modal/shareable-url-modal.js:
```diff
@@ -53,7 +55,11 @@ const ShareableUrlModal = ({ onToggleModal, visible }) => {
   const [showCopied, setShowCopied] = useState(false);
   const [compatibilityData, setCompatibilityData] = useState({});
   const [canUseShareableUrls, setCanUseShareableUrls] = useState(true);
```
nit

```suggestion
const [isShareableUrlsEnabled, setIsShareableUrlsEnabled] = useState(true);
```

Huongg replied:
hmm i don't think it's about enabled or not, how about `[isCompatible, setIsCompatible]` ? wdyt @rashidakanchwala

rashidakanchwala commented on src/components/shareable-url-modal/shareable-url-modal.js:
```diff
@@ -111,6 +167,18 @@ const ShareableUrlModal = ({ onToggleModal, visible }) => {
       if (request.ok) {
         setResponseUrl(response.url);
         setDeploymentState('success');
+        setToDisplayPublishedView();
```
Can u pls explain the purpose of this?

Huongg replied:
hey so this is to make sure if the request is okay, we should run the check to see if localStorage is existed then update it the local state accordingly. it's now called `displayPublishedView`

rashidakanchwala commented on src/components/shareable-url-modal/shareable-url-modal.js:
```diff
@@ -82,6 +88,43 @@ const ShareableUrlModal = ({ onToggleModal, visible }) => {
     fetchPackageCompatibility();
   }, []);
 
+  const setToDisplayPublishedView = () => {
```
nit 
```suggestion
  const displayPublishedView = () => {
```

rashidakanchwala commented on src/components/shareable-url-modal/shareable-url-modal.js:
```diff
@@ -82,6 +88,43 @@ const ShareableUrlModal = ({ onToggleModal, visible }) => {
+    }
+  };
+
+  const setToDisplayMainViewWithPopulatedContent = () => {
```
nit 
```suggestion
  const displayMainViewWithStoredContent = () => {
```

jitu5 commented:
@Huongg When I was testing locally, For the first time after successful deployment I got below screen. Is it expected to have no close or cancel button on success screen ? 

<img width="967" alt="Screenshot 2024-05-22 at 10 11 34 a m" src="https://github.com/kedro-org/kedro-viz/assets/38945204/76f444f9-a18c-4626-8e64-cb619834c468">

Also after refresh I opened publish popup again I got below screen, here also we don't have no close or cancel button. Thanks. 
<img width="818" alt="Screenshot 2024-05-22 at 10 17 41 a m" src="https://github.com/kedro-org/kedro-viz/assets/38945204/939efda2-0590-4486-94f2-a1cd8fab13b0">

Huongg commented:
> @Huongg When I was testing locally, For the first time after successful deployment I got below screen. Is it expected to have no close or cancel button on success screen ?
> 
> <img alt="Screenshot 2024-05-22 at 10 11 34 a m" width="967" src="https://private-user-images.githubusercontent.com/38945204/332737275-76f444f9-a18c-4626-8e64-cb619834c468.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTYzNzUyODEsIm5iZiI6MTcxNjM3NDk4MSwicGF0aCI6Ii8zODk0NTIwNC8zMzI3MzcyNzUtNzZmNDQ0ZjktYTE4Yy00NjI2LThlNjQtY2I2MTk4MzRjNDY4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTIyVDEwNDk0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ3ZWQxMzE2ZWU4YTYxZmJmMDlhMDNjNjE2YzA2NDMwY2VjMTY0ZmQ4NDE0Njc2MGEzYzAwMGZjOWRkY2RlMWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.PgIKC6v0FsGDwuKT_2I0rRt94Ah5ItqCDZNq_Vqw9eE">
> Also after refresh I opened publish popup again I got below screen, here also we don't have no close or cancel button. Thanks. <img alt="Screenshot 2024-05-22 at 10 17 41 a m" width="818" src="https://private-user-images.githubusercontent.com/38945204/332738193-939efda2-0590-4486-94f2-a1cd8fab13b0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTYzNzUyODEsIm5iZiI6MTcxNjM3NDk4MSwicGF0aCI6Ii8zODk0NTIwNC8zMzI3MzgxOTMtOTM5ZWZkYTItMDU5MC00NDg2LTk0ZjItYTFjZDhmYWIxM2IwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTIyVDEwNDk0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTEzNGU2MDkzYTY2NmNhM2FmYTBhYWVkMzAyZTIxNjg2Yjg0OTFiZTczNzVjNDgzYWY2YjMxZDE4N2NhNmFjZWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.0Eb-Z9MvFVc1GetA5jRpqk90zNxGXgI1VJGSPpUJGSA">

hey @jitu5  yes it is expected, according to the design [here](https://www.figma.com/file/3kSpvIO1veLKfy9qHxuXwF/Kedro-WIP?type=design&node-id=1411-87299&mode=design&t=bSwcXxbq95GPPfud-4)

Huongg commented:
> @Huongg As we discuss we need to refactor or break down shareable-url-modal.js component as its getting bigger and difficult to read.

hey agree, I've done some refactoring to separate the components and simplify it's logic so we don't return null too much. Please take a look when you have time @rashidakanchwala @jitu5

stephkaiser commented:
thanks @Huongg!

Design QA notes below:
- padding between modal title and 'Prerequisites' text should be 24px
- Text fixes - 
-- Prerequisite(s) is missing an 's'
-- both Prerequisites and Disclaimer doesn't have the semicolon :
-- both Prerequisites and Disclaimer subtitles have regular font weight, in white-0 colour (#FFFFFF 100%)
-- Body text under Prerequisites and Disclaimer colour should be white-900 (#C6CACE 100%)
-- 'Enter the required information' should be 'Please enter the required information below.'
-- 'Hosting platform' empty state text has a different colour to the other two text inputs, it looks a little blue. Empty state text should be black-500 (#424242 100%)
-- Text field labels (Hosting platform, Bucket name, Endpoint URL) should be in black-0 (#989898 100%)
-- Republish text should be "Republish Kedro-Viz to push new updates, or publish and host Kedro-Viz with a new link."
- Dataset preview toggle text and toggle component is not aligned, text is slightly higher
- Endpoint URL (i) tooltip - position is too close to the (i) icon. Padding is too much on the top + bottom, and size is smaller
- Loading modal & Error modal - padding is too much above the title, should be 48px
- Copy link button - 
-- On hover state button turns white, and on click only the 'Copied!' tooltip appears above
-- After clicking on the button, the size and shape of the button changes
- Multiple hosted link dropdown next to link - padding should be 20px
- Light Mode - dropdown menus should be white, they are currently grey

Huongg commented:
Hey @stephkaiser thank you, I believed I have addressed all your design comments, let me know if you spotted anything else

- [ ] padding between modal title and 'Prerequisites' text should be 24px
- [ ] Text fixes -
- [ ] -- Prerequisite(s) is missing an 's'
- [ ] -- both Prerequisites and Disclaimer doesn't have the semicolon :
- [ ] -- both Prerequisites and Disclaimer subtitles have regular font weight, in white-0 colour (#FFFFFF 100%)
- [ ] -- Body text under Prerequisites and Disclaimer colour should be white-900 (#C6CACE 100%)
- [ ] -- 'Enter the required information' should be 'Please enter the required information below.'
- [ ] -- 'Hosting platform' empty state text has a different colour to the other two text inputs, it looks a little blue. Empty state text should be black-500 (#424242 100%)
- [ ] -- Text field labels (Hosting platform, Bucket name, Endpoint URL) should be in black-0 (#989898 100%)
- [ ] -- Republish text should be "Republish Kedro-Viz to push new updates, or publish and host Kedro-Viz with a new link."
- [ ] Dataset preview toggle text and toggle component is not aligned, text is slightly higher
- [ ] Endpoint URL (i) tooltip - position is too close to the (i) icon. Padding is too much on the top + bottom, and size is smaller
- [ ] Loading modal & Error modal - padding is too much above the title, should be 48px
- [ ] Copy link button -
- [ ] -- On hover state button turns white, and on click only the 'Copied!' tooltip appears above
- [ ] -- Currently, after clicking on the button the size and shape of the button changes, it should stay the same
- [ ] Multiple hosted link dropdown (next to link) - left side padding should be 20px
- [ ] Light Mode - dropdown menus should be white, they are currently grey

stephkaiser commented:
Nice work! thank you very much @Huongg!

To clarify the one in bold - **Multiple hosted link dropdown (next to link) - left side padding should be 20px** -
I was referring to this dropdown in the retrieving url modal, the left side padding of the dropdown should be 20px so that the text aligns with the text in the dropdown menu. 
<img width="250" alt="Screenshot 2024-05-28 at 16 14 26" src="https://github.com/kedro-org/kedro-viz/assets/31114028/24efe06a-e897-464f-a960-2f46d00710d6">

Also, the same dropdown menus in light mode have a black border, can we remove this please?
<img width="250" alt="Screenshot 2024-05-28 at 16 19 45" src="https://github.com/kedro-org/kedro-viz/assets/31114028/9eb1c82f-3c00-4f72-b247-eaea189cd54a">
<img width="250" alt="Screenshot 2024-05-28 at 16 19 57" src="https://github.com/kedro-org/kedro-viz/assets/31114028/e3696b3f-0c6f-48e3-b409-7a7ad6288dc9">

Thank you for fixing the "padding is too much above the title, should be 48px". Can we do the same for the right section (with the form) as well so that it is also aligned with the title?
<img width="500" alt="Screenshot 2024-05-28 at 16 30 29" src="https://github.com/kedro-org/kedro-viz/assets/31114028/b1b6726a-a961-49ae-9221-212debfbbaba">

And last one - would it be possible to move the 'or publish...' text to the next line so that it reads better? 
_Implementation_ - 
<img width="500" alt="Screenshot 2024-05-28 at 16 28 33" src="https://github.com/kedro-org/kedro-viz/assets/31114028/96a3ab4c-4101-4d70-91c3-51d0572967a7">
_Design_ - 
<img width="500" alt="Screenshot 2024-05-28 at 17 24 39" src="https://github.com/kedro-org/kedro-viz/assets/31114028/172a7c77-5b0e-4528-9db3-e5e7c2f7c521">

Thank you!

jitu5 commented on src/components/shareable-url-modal/shareable-url-modal.js:
```diff
@@ -111,6 +149,18 @@ const ShareableUrlModal = ({ onToggleModal, visible }) => {
         setDeploymentState('success');
+        displayPublishedView();
+
+        // const hostingPlatformVal = {};
+        // if (hostingPlatforms.hasOwnProperty(inputValues.platform)) {
+        //   hostingPlatformVal[inputValues.platform] = { ...inputValues };
+        // }
+        // saveLocalStorage(localStorageSharableUrl, hostingPlatformVal);
+        // const newState = {
+        //   ...hostingPlatformLocalStorageVal,
+        //   ...hostingPlatformVal,
+        // };
+        // setHostingPlatformLocalStorageVal(newState);
```
Please remove the comments if not needed.

Huongg replied:
once Step approves the design I'll move the logic back here

Huongg replied:
done

jitu5 commented:
@Huongg I see underline on hover republish button, is it expected ?
<img width="708" alt="Screenshot 2024-05-31 at 11 32 51 a m" src="https://github.com/kedro-org/kedro-viz/assets/38945204/c410f621-1ebd-43df-99d9-f12815bf50b4">

jitu5 commented on src/components/shareable-url-modal/shareable-url-modal.js:
```diff
@@ -98,6 +124,18 @@ const ShareableUrlModal = ({ onToggleModal, visible }) => {
+    setHostingPlatformLocalStorageVal(newState);
+
     try {
       const request = await fetch('/api/deploy', {
```
Optional: we can move all the API requests to common place.

jitu5 commented on src/components/shareable-url-modal/shareable-url-modal.js:
```diff
@@ -98,6 +124,18 @@ const ShareableUrlModal = ({ onToggleModal, visible }) => {
     setDeploymentState('loading');
     setIsLoading(true);
 
+    // this logic is here to test locally without publishing anything
+    const hostingPlatformVal = {};
+    if (hostingPlatforms.hasOwnProperty(inputValues.platform)) {
+      hostingPlatformVal[inputValues.platform] = { ...inputValues };
+    }
+    saveLocalStorage(localStorageSharableUrl, hostingPlatformVal);
+    const newState = {
+      ...hostingPlatformLocalStorageVal,
+      ...hostingPlatformVal,
+    };
+    setHostingPlatformLocalStorageVal(newState);
```
Nit: This part about localStorage can be moved into a separate function.

Huongg replied:
lets discuss about this when you're back 👍

stephkaiser commented:
> @Huongg I see underline on hover republish button, is it expected ? <img alt="Screenshot 2024-05-31 at 11 32 51 a m" width="708" src="https://private-user-images.githubusercontent.com/38945204/335545323-c410f621-1ebd-43df-99d9-f12815bf50b4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTcxNTIwMTIsIm5iZiI6MTcxNzE1MTcxMiwicGF0aCI6Ii8zODk0NTIwNC8zMzU1NDUzMjMtYzQxMGY2MjEtMWViZC00M2RmLTk5ZDktZjEyODE1YmY1MGI0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTMxVDEwMzUxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRmNDEyMTU1Y2NmZDc3OWNiMjk2YjdjZjQzOTAzZTg3NGM4N2E3YWQ3Yjk2NDc3ODM4ZGI3MTQ4Y2I4MDdmMzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.5eI4czsN0sZu_RKTtX9VQudmdj0VEBLpzu2Yq9NjCTE">

thanks Jitendra, the underline for the hover state on the button is fine thanks!

Huongg commented:
thanks @stephkaiser I've just addressed the rest of your comment Step, Let me know if you're happy with all the design

stephkaiser commented:
> thanks @stephkaiser I've just addressed the rest of your comment Step, Let me know if you're happy with all the design

looks perfect, no more comments from me. Thank you so much Huong! 🙏

jitu5 reviewed (approved):
Amazing work Huong!

Huongg commented on src/components/shareable-url-modal/shareable-url-modal.js:
```diff
@@ -53,7 +55,11 @@ const ShareableUrlModal = ({ onToggleModal, visible }) => {
+  const [hostingPlatformLocalStorageVal, setHostingPlatformLocalStorageVal] =
+    useState(loadLocalStorage(localStorageSharableUrl) || {});
+  const [populatedContentKey, setPopulatedContentKey] = useState(undefined);
+  const [toggleValue, setTogleValue] = useState(false);
```
for now it's just an empty function as i dont know what needs to go in here. It will be taken over by @jitu5 once I merge to main. 

@jitu5 do we need  isPreviewEnabled with the logic you have from your side?

jitu5 replied:
Yes, setTogleValue/isPreviewEnabled will be needed for disable preview toggle button.

ravi-kumar-pilla replied:
can we rename this to isPreviewEnabled, setIsPreviewEnabled ?

Huongg replied:
yup sure 👍

Huongg replied:
done

Huongg commented on src/components/shareable-url-modal/shareable-url-modal.js:
```diff
@@ -53,7 +55,11 @@ const ShareableUrlModal = ({ onToggleModal, visible }) => {
   const [compatibilityData, setCompatibilityData] = useState({});
   const [canUseShareableUrls, setCanUseShareableUrls] = useState(true);
-  const [isDisclaimerViewed, setIsDisclaimerViewed] = useState(false);
+  const [showPublishedView, setShowPublishedView] = useState(false);
```
i think it's more about to show the publish view or not, so I'd prefer to keep it how it is

ravi-kumar-pilla commented on RELEASE.md:
```diff
@@ -7,6 +7,11 @@ Please follow the established format:
 # Next release
 
+## Major features and improvements
+- Display previously published URLs. (#1907)
```
Display URLs that were published ?

Huongg replied:
"Display published URLs" ?

ravi-kumar-pilla commented on src/components/shareable-url-modal/main-view/main-view.js:
```diff
@@ -0,0 +1,190 @@
+  KEDRO_VIZ_PUBLISH_DOCS_URL,
+} from '../../../config';
+
+const renderTextContent = (toggleValue, setTogleValue) => {
```
(toggleValue, setTogleValue) - if this is only for previews, can we rename to (isPreviewEnabled, setIsPreviewEnabled) ?

ravi-kumar-pilla commented on src/utils/index.js:
```diff
@@ -227,6 +227,18 @@ export async function fetchPackageCompatibilities() {
   return request;
 }
 
+export async function retrieveHostingPlatformData(inputValues) {
```
As this method is used to deploy (POST), can we rename this to publishViz or deployViz ?

Huongg replied:
hey yeah good suggestion, i couldn't think of any good name for it, thank you

ravi-kumar-pilla commented on src/components/shareable-url-modal/published-view/published-view.js:
```diff
@@ -0,0 +1,91 @@
+  platform,
+  showCopied,
+}) => {
+  const platformsKeys = Object.keys(hostingPlatformLocalStorageVal);
```
```suggestion
  const hostedPlatformKeys = Object.keys(hostingPlatformLocalStorageVal);
```

ravi-kumar-pilla replied:
does - Object.keys(hostingPlatformLocalStorageVal); return ["aws", "azure", "gcp"] from the localStorage -
{
   aws: {
       bucket_name: ...,
       platform: 'aws',
       endpoint: '....',
   },
   gcp: {....},
   azure: {....}
}
?

Also, in localStorage do we have a parent to the above object ? Like do we store it under kedro-viz ?
kedro-viz: {
....
hostedPlatforms: {
aws: {
       bucket_name: ...,
       platform: 'aws',
       endpoint: '....',
   },
   gcp: {....},
   azure: {....}
}
}

Huongg replied:
hey so 

- does - Object.keys(hostingPlatformLocalStorageVal); return ["aws", "azure", "gcp"] from the localStorage ? 
yes it's correct
- Also, in localStorage do we have a parent to the above object ? Like do we store it under kedro-viz ? 
yes so this one live under `KedroViz-sharable-url'`, like other things we stored in localStorage
```
export const localStorageName = 'KedroViz';
export const localStorageFlowchartLink = 'KedroViz-link-to-flowchart';
export const localStorageMetricsSelect = 'KedroViz-metrics-chart-select';
export const localStorageRunsMetadata = 'KedroViz-runs-metadata';
export const localStorageSharableUrl = 'KedroViz-sharable-url';
```

ravi-kumar-pilla commented on src/components/shareable-url-modal/published-view/published-view.js:
```diff
@@ -0,0 +1,91 @@
+  showCopied,
+}) => {
+  const platformsKeys = Object.keys(hostingPlatformLocalStorageVal);
+  const platformsVal = Object.values(hostingPlatformLocalStorageVal);
```
```suggestion
  const hostedPlatformValuesFromLocalStorage = Object.values(hostingPlatformLocalStorageVal);
```

ravi-kumar-pilla commented on src/components/shareable-url-modal/published-view/published-view.js:
```diff
@@ -0,0 +1,91 @@
+
+  const url = platform
+    ? hostingPlatformLocalStorageVal[platform]['endpoint']
+    : platformsVal[0]['endpoint'];
```
`published-view` only appears after atleast one successful publish ? If not, does platformsVal always contain atleast 1 element ?

Huongg replied:
in the published-view, yes it always contains at least 1 element as there's a condition to only render the publishedView if the length of value we get from localStorage is greater than 1

ravi-kumar-pilla commented on src/components/shareable-url-modal/published-view/published-view.js:
```diff
@@ -0,0 +1,91 @@
+        <div className="shareable-url-modal__content-title">
+          Publish and Share Kedro-Viz
+        </div>
+        {platformsKeys.length === 1 ? (
```
I have published once on AWS and the next time, I would like to publish on Azure. Now, my localStorage contains 1 key which makes this condition true. How can I publish on Azure ? I do not see a link on `publish and host Kedro-Viz with a new link.` . Am I missing something ?

``` <div className="shareable-url-modal__published-action">
        <p className="shareable-url-modal__published-action-text">
          Republish Kedro-Viz to push new updates,
          <br />
          or publish and host Kedro-Viz with a new link.
        </p>
        <Button mode="secondary" onClick={onRepublishClick} size="small">
          Republish
        </Button>
      </div>
  ```

ravi-kumar-pilla commented on src/components/shareable-url-modal/shareable-url-modal.js:
```diff
@@ -143,341 +176,93 @@ const ShareableUrlModal = ({ onToggleModal, visible }) => {
-      {!isDisclaimerViewed && canUseShareableUrls ? (
-        renderDisclaimerContent()
+      {!isCompatible ? (
+        <CompatibilityView onClick={handleModalClose} />
```
[Nit] This seems to be a view for compatibility error. Can we rename it to be CompatibilityErrorView ?

Huongg replied:
yeah good suggestion, will do that

ravi-kumar-pilla commented on src/components/shareable-url-modal/published-view/published-view.js:
```diff
@@ -0,0 +1,91 @@
+            <Dropdown
+              defaultText={
+                (platform && filteredPlatforms[platform]) ||
+                Object.values(filteredPlatforms)[0]
```
since we store localStorage as -
```
{
   aws: {
       bucket_name: ...,
       platform: 'aws',
       endpoint: '....',
   },
   gcp: {....},
   azure: {....}
}
```
If my previous deployments were in the order - azure, aws, gcp - this will fill up azure for my 4th deployment as default ? Let's say I deployed with aws for my 4th deployment, will the localStorage keys order according to the latest deployment ? After 4th deployment will the keys be ordered like aws, gcp, azure ?

To summarize will filteredPlatforms[0] always points to last deployment ?

Huongg replied:
at the moment, the order of the list will stay the same, the first item of the list is not your lates deployment but the first one you deployed. Good question, let see what @stephkaiser think on this

ravi-kumar-pilla commented:
Hi @Huongg , Apologies for the delay in reviewing the PR. Most of the code looks good. I had a first pass and left few comments and questions. Please have a look. Thank you

Huongg commented on src/components/shareable-url-modal/shareable-url-modal.js:
```diff
@@ -143,341 +176,93 @@ const ShareableUrlModal = ({ onToggleModal, visible }) => {
       hasPlatform: false,
       hasEndpoint: false,
     });
+    displayPublishedView();
```
hey so originally there are 2 main places where it checks to render the publishedView or not, first render or when it deployed successfully. However lets say when the user close the modal even they havent published anything yet, we still need to run  `displayPublishedView()` again where it checks localStorage existed or not, if it does => set the publishedView again, if not, nothing happened. Does it make sense?

ravi-kumar-pilla commented on src/components/shareable-url-modal/shareable-url-modal.js:
```diff
@@ -82,6 +72,43 @@ const ShareableUrlModal = ({ onToggleModal, visible }) => {
     fetchPackageCompatibility();
   }, []);
 
+  const displayPublishedView = () => {
```
```suggestion
  const setStateForPublishedView = () => {
```

ravi-kumar-pilla replied:
I realized we are not displaying any UI but rather setting/updating state. It would be better to rename this to convey the same

ravi-kumar-pilla commented on src/components/shareable-url-modal/shareable-url-modal.js:
```diff
@@ -82,6 +72,43 @@ const ShareableUrlModal = ({ onToggleModal, visible }) => {
+    }
+  };
+
+  const displayMainViewWithPublishedContent = () => {
```
```suggestion
  const setStateForMainViewWithPublishedContent = () => {
```

Huongg commented:
> Hi @Huongg , Thanks for addressing all the comments. There are some minor observations and I would like to know what the team thinks @stephkaiser @jitu5 -
> 
> 1. When closing the modal of published view - It changes the form before closing the modal (minor thing but seems like something is changing on the screen)
> 2. When closing the modal of main content from published view - It changes the modal to publish view before closing (minor thing but seems like something is changing on the screen)
> 3. When I click on republish, the form data gets filled based on the selected provider, but if I change the dropdown source, even if the provider is in localStorage, the form is not populated.
> 
> Let me know if you need more info. Thank you
> 
> ![minor_close_issue](https://private-user-images.githubusercontent.com/87735534/337292582-fff0a66e-d314-41e6-b653-250545590a82.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc2ODIzNTEsIm5iZiI6MTcxNzY4MjA1MSwicGF0aCI6Ii84NzczNTUzNC8zMzcyOTI1ODItZmZmMGE2NmUtZDMxNC00MWU2LWI2NTMtMjUwNTQ1NTkwYTgyLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA2VDEzNTQxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM1MzdiN2I3Y2JjYmU1OTU5ZTM3ZTJmOTkyOTBlYzI2NWIzNTkwNDQ2ZjBhNTVjNmNmZmI5YThhNGJlNmM2NGMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.AEb4ReNTtudPtvJbPMrEothH-ABpWKL5Rhfm0x1FW50) [ ![minor_close_issue](https://private-user-images.githubusercontent.com/87735534/337292582-fff0a66e-d314-41e6-b653-250545590a82.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc2ODIzNTEsIm5iZiI6MTcxNzY4MjA1MSwicGF0aCI6Ii84NzczNTUzNC8zMzcyOTI1ODItZmZmMGE2NmUtZDMxNC00MWU2LWI2NTMtMjUwNTQ1NTkwYTgyLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA2VDEzNTQxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM1MzdiN2I3Y2JjYmU1OTU5ZTM3ZTJmOTkyOTBlYzI2NWIzNTkwNDQ2ZjBhNTVjNmNmZmI5YThhNGJlNmM2NGMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.AEb4ReNTtudPtvJbPMrEothH-ABpWKL5Rhfm0x1FW50) ](https://private-user-images.githubusercontent.com/87735534/337292582-fff0a66e-d314-41e6-b653-250545590a82.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc2ODIzNTEsIm5iZiI6MTcxNzY4MjA1MSwicGF0aCI6Ii84NzczNTUzNC8zMzcyOTI1ODItZmZmMGE2NmUtZDMxNC00MWU2LWI2NTMtMjUwNTQ1NTkwYTgyLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA2VDEzNTQxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM1MzdiN2I3Y2JjYmU1OTU5ZTM3ZTJmOTkyOTBlYzI2NWIzNTkwNDQ2ZjBhNTVjNmNmZmI5YThhNGJlNmM2NGMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.AEb4ReNTtudPtvJbPMrEothH-ABpWKL5Rhfm0x1FW50) [ ](https://private-user-images.githubusercontent.com/87735534/337292582-fff0a66e-d314-41e6-b653-250545590a82.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc2ODIzNTEsIm5iZiI6MTcxNzY4MjA1MSwicGF0aCI6Ii84NzczNTUzNC8zMzcyOTI1ODItZmZmMGE2NmUtZDMxNC00MWU2LWI2NTMtMjUwNTQ1NTkwYTgyLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA2VDEzNTQxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM1MzdiN2I3Y2JjYmU1OTU5ZTM3ZTJmOTkyOTBlYzI2NWIzNTkwNDQ2ZjBhNTVjNmNmZmI5YThhNGJlNmM2NGMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.AEb4ReNTtudPtvJbPMrEothH-ABpWKL5Rhfm0x1FW50)

hey so for point 1 and 2, it has been a behaviour of the form, if we think this is not right then we can create a separate issue for it. And for point 3, when you're in the form, it does not read from the localStorage anymore so when you change to a different platform, it wont be re-populated

Huongg commented:
FYI: link to the bug report for @ravi-kumar-pilla's comment 1 and 2: https://github.com/kedro-org/kedro-viz/issues/1936

ravi-kumar-pilla reviewed (approved):
Thank you @Huongg ... looks great 💯
